custom_components/tuya_local/devices/duux_whisper_flex_ultimate_fan.yaml

```diff
@@ -0,0 +1,96 @@
+name: Portable Fan
+products:
+  - id: cjsa4k7tqbpuyq3l
+    name: Duux Whisper Flex Ultimate
+primary_entity:
+  entity: fan
+  translation_only_key: fan_with_presets
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 2
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: "0"
+          value: normal
+        - dps_val: "1"
+          value: nature
+        - dps_val: "2"
+          value: sleep
+    - id: 3
+      type: integer
+      name: speed
+      mapping:
```

Comment:
You should use range rather than a scale for fan speed.  The speed is now auto-scaled using https://developers.home-assistant.io/blog/2023/12/05/scaling-utils/